eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxOut.hs

```diff
@@ -173,15 +168,17 @@ viewCompactTxOut ::
 viewCompactTxOut txOut = case txOut of
   TxOutCompact' addr val -> (addr, val, SNothing)
   TxOutCompactDH' addr val dh -> (addr, val, SJust dh)
-  TxOut_AddrHash28_AdaOnly stakeRef addr28Extra adaVal
-    | Just addr <- decodeAddress28 stakeRef addr28Extra ->
-        (compactAddr addr, injectCompact adaVal, SNothing)
-    | otherwise -> error addressErrorMsg
-  TxOut_AddrHash28_AdaOnly_DataHash32 stakeRef addr28Extra adaVal dataHash32
-    | Just addr <- decodeAddress28 stakeRef addr28Extra
-    , Just dh <- decodeDataHash32 dataHash32 ->
-        (compactAddr addr, injectCompact adaVal, SJust dh)
-    | otherwise -> error addressErrorMsg
+  TxOut_AddrHash28_AdaOnly stakeRef addr28Extra adaVal ->
+    let
+      !addr = decodeAddress28 stakeRef addr28Extra
```

Comment:
Making the result of this function strict is actually a bad idea. 
When one does a pattern match on the `TxOut`, a user doesn't necessarily needs all the parts of it uncompacted.
```suggestion
      addr = decodeAddress28 stakeRef addr28Extra
```

Reply:
I tried to keep the strictness as it was before the refactoring, but we can get rid of this.

Reply:
Ok, I think I see your point now, previously it used to evaluate the result of the decoding only when you pattern matched on the `Maybe`, but now it evaluates it in this function, which is different. I'll get rid of it.